### DISCUSSION TITLE
整了个 GitHub copilot

### DISCUSSION THREAD

**Discussion by mark9804** [Arondight/Adachi-BOT#748] [General]
不过我没用 webpack 和 vue 脚手架（不会），也没用 vue template，不知道在这种情况下还能不能香得起来

**Comment by mark9804** [comment]
wtm的，这离大谱，注释都能补……（PS：这行注释我从没写过）
<img width="1112" alt="image" src="https://user-images.githubusercontent.com/9006264/159634386-c67e631a-165d-4a3c-968d-0a8f835b8be6.png">

**Reply by Arondight** [reply]
这玩意儿是需要申请的？

**Reply by mark9804** [reply]
要申请，我上个月 13 号申请的

**Comment by mark9804** [comment]
试了一下，能补全的东西挺多的，我试了一下python，js都可以补，拿R做结构方程模型都能给我补上，只是猜错了我的数据集名称

通常情况下copilot只会补全最多两行左右，但是如果你的代码有重复性比较高的地方，它会一次性补完整个函数体或者循环体，不过这种情况下我一般会考虑要不要继续抽象一下，不会接受它的建议

总结一下第一天体验，就一个字离谱

**Reply by mark9804** [reply]
第一天：牛逼
第二天：牛逼

<img width="700" alt="image" src="https://user-images.githubusercontent.com/9006264/159924053-cf523d51-3f30-4a18-8d8e-5426c44c4405.png">

**Reply by Arondight** [reply]
我也申请了，到时候跟同事装装逼

**Reply by mark9804** [reply]
公司环境会不会不让用这种东西，毕竟可能还是涉及一些秘密的

**Reply by Arondight** [reply]
> 公司环境会不会不让用这种东西，毕竟可能还是涉及一些秘密的

自个写代码演示一下，开会吹吹牛，不会用在公司的项目上的

**Comment by mark9804** [comment]
> 通常情况下copilot只会补全最多两行左右

我收回这句话（逻辑和用法不对，但是补的还行）

<img width="607" alt="image" src="https://user-images.githubusercontent.com/9006264/160221321-8069b5a8-f4ba-40e1-b0b9-41127f11ac68.png">

<img width="984" alt="image" src="https://user-images.githubusercontent.com/9006264/160221464-64f21e7e-f242-41f4-82b9-4ed56d43d1f1.png">

<img width="627" alt="image" src="https://user-images.githubusercontent.com/9006264/160227827-e69af574-8f93-445c-bee5-55e8d3daafc1.png">

**Comment by mark9804** [comment]
是不是有人在 GitHub 上面写小说
<img width="203" alt="image" src="https://user-images.githubusercontent.com/9006264/160749082-e47fab9c-c4f8-4284-aa8b-88ec956de488.png">
<img width="178" alt="image" src="https://user-images.githubusercontent.com/9006264/160749087-fac332c1-e126-49e1-8be0-d2cf3bfefed4.png">
<img width="234" alt="image" src="https://user-images.githubusercontent.com/9006264/160749104-7f08ffd6-bf4a-48a5-906d-3d5b1c551837.png">

**Comment by Arondight** [comment]
![图片](https://user-images.githubusercontent.com/13673702/161973993-8d2ae57a-aeee-412f-93ca-c099d70b8f12.png)

![图片](https://user-images.githubusercontent.com/13673702/161974044-d33804fe-2630-4fae-8848-75c586089534.png)

牛皮

**Reply by Arondight** [reply]
![图片](https://user-images.githubusercontent.com/13673702/161976053-65f144b2-f032-47b6-9216-a7140c05e70e.png)

给我补全的注释，虽然 bot 参数理解错了，而且返回值没有覆盖所有情况

**Reply by mark9804** [reply]
我用了一段时间之后，感觉它是通过理解程序结构的方式工作的，假如之前有人写过和你差不多的程序结构，那他就会给你把这段代码补全出来

本质上它还是不会“自己写代码”的，不如说更像是一个自动 stackoverflow 然后复制粘贴的机器

**Comment by mark9804** [comment]
?
<img width="686" alt="image" src="https://user-images.githubusercontent.com/9006264/162560864-49c23387-6bc7-42bf-a713-3a50a793d21f.png">

**Comment by mark9804** [comment]
copilot 会自动翻译，而且会根据对之前译文的改正（人名之类的）自动修正后面的补全建议，已经在给组里的翻译推荐了

<img width="415" alt="image" src="https://user-images.githubusercontent.com/9006264/201477291-d3414589-96c0-4938-b48a-ce0ef43f4083.png">
<img width="330" alt="image" src="https://user-images.githubusercontent.com/9006264/201477293-5bb1e20f-c0fc-4d02-992c-0380f4cdeb99.png">

**Reply by Arondight** [reply]
组里写了个程序调用有道翻译的 po ，感觉还快一点

**Reply by mark9804** [reply]
> 有道翻译

API 有次数限制吗，想了想也是个不错的办法，就是特殊字符得预处理掉

**Reply by Arondight** [reply]
这种公开的 api 哪有限制次数的，那同行可笑嘻了

**Comment by BTMuli** [comment]
之前申请了好几次不通过，刚刚又申请了一次，查了下才知道不能开代理，图片我传的学信网，终于通过了，哭死
![image](https://user-images.githubusercontent.com/72692909/201502120-8c1e88f4-31d4-4ab0-a144-88ca13356e12.png)

**Reply by mark9804** [reply]
教育认证状态不是和 GitHub 账号绑定的吗？

**Reply by BTMuli** [reply]
并不是，我当初注册账号邮箱用的QQ好像，后面改成outlook了，一开始用学校邮箱可能会绑定吧Σ(|||▽||| )

**Reply by BTMuli** [reply]
用完了，真的强👍

本来以为类似那种 enableUser disableUser的直接写完函数名就生成代码已经够厉害的了

直到我写了个 //弹窗 

就写完一个 func 了

你们程序员代码写这么多，一定知道怎么实现的吧（

**Reply by mark9804** [reply]
copilot 应该是通过理解程序结构来工作的，假如别人写过差不多的结构那么它就会给你补出来，但假如你在做一些比较新的，别人不一定东西就会发现它帮倒忙的时候比较多。我觉得他对我有大用反而挺悲哀的（）

但是 copilot 居然能自动翻译属实我没想到，这都喂了什么东西进去

不过现在在赶教授那边的东西（教授：界面看起来没变化啊，你这周在摸鱼吧），对于赶工人来说确实有用……

**Comment by mark9804** [comment]
AI 代码补全插件 kite 破产（总共50万用户，其中白嫖用户50万），开源大部分代码，其中包括数据驱动的 Python 类型推理引擎、Python 公共包分析器、桌面软件、适配各种编辑器的插件、Github 爬虫和分析器等

https://github.com/kiteco
